core/src/main/java/com/redhat/runtimes/inventory/auth/ConsoleIdentityProvider.java

```diff
@@ -0,0 +1,168 @@
+                                 * IOException is thrown when the connection between us and RBAC is reset during an RBAC call execution.
+                                 * ConnectTimeoutException is thrown when RBAC does not respond at all to our call.
+                                 */
+                                .onFailure(
+                                    failure ->
+                                        failure.getClass() == IOException.class
+                                            || failure.getClass() == ConnectTimeoutException.class)
+                                .retry()
+                                .withBackOff(initialBackOff, maxBackOff)
+                                .atMost(maxRetryAttempts)
+                                // After we're done retrying, an RBAC server call failure will cause
+                                // an authentication failure
+                                .onFailure()
+                                .transform(
+                                    failure -> {
+                                      throw new AuthenticationFailedException(
+                                          "RBAC authentication call failed", failure);
+                                    })
+                                // Otherwise, we can finish building the QuarkusSecurityIdentity and
+                                // return the result
+                                .onItem()
+                                .transform(
+                                    rbacRaw -> {
+                                      if (rbacRaw.canRead("inventory", "hosts")) {
+                                        builder.addRole(RBAC_READ_HOSTS);
+                                      }
+                                      if (rbacRaw.canWrite("inventory", "hosts")) {
+                                        builder.addRole(RBAC_WRITE_HOSTS);
+                                      }
+                                      return builder.build();
+                                    });
+                          } else {
+                            Log.warnf(
+                                "Unprocessed identity found. type: %s and name: %s",
+                                identity.type, identity.getName());
+                            return Uni.createFrom().failure(new AuthenticationFailedException());
+                          }
+                        })
+                    // A failure will cause an authentication failure
+                    .onFailure()
+                    .transform(
+                        throwable -> {
+                          Log.error("Error while processing identity", throwable);
+                          return new AuthenticationFailedException(throwable);
+                        }));
+  }
+
+  private static ConsoleIdentity getRhIdentityFromString(String xRhIdHeader) {
```

Comment:
Can this be made default access & unit tested?

Reply:
Added tests under com.redhat.runtimes.inventory.auth.RhIdentityParsingTest using some example RhId headers from notifications-backend.